### ISSUE TITLE
`--set` with `.dns` doesn't always seem to work 

### ISSUE THREAD

**Issue by pmalek** [lima-vm/lima#1940]
### Description

I use the following set of flags

```
'--set=.cpus = 6 | .memory = "10GiB" | .arch = "aarch64" | .vmType = "vz"  | .dns = []  | .dns += "192.168.1.250"'
```

Recently (with b5aa0dd) discovered that the dns setting doesn't always work as expected. I'm not sure what should be the effect of that setting but sometimes it does put `192.168.1.250` in `/etc/resolv.conf` and sometimes doesn't.

What should be the effect of that setting?

---

```
limactl -v
limactl version HEAD-b5aa0dd
```

**Comment by balajiv113** [comment]
@pmalek Which template are you using ?? 
Is it Alpine or Ubuntu ??

**Comment by neowu** [comment]
i encountered same behavior, after digging little into source code

it looks like, if vmType == VZ, then it's always using args.SlirpDNS (which is gatewayIp) to generate user-data
and bypass DNS settings in lima.yml, then output user-data like

```
manage_resolv_conf: true

resolv_conf:
  nameservers:
  - 192.168.5.2
```

not sure whether it's intended behavior.

from the source 
https://github.com/lima-vm/lima/blob/391133d0b37532d0ac94872a81c3754f92c2531c/pkg/cidata/cidata.go#L268
always go into first if condition branch
```
	if firstUsernetIndex != -1 || *y.VMType == limayaml.VZ {
		args.DNSAddresses = append(args.DNSAddresses, args.SlirpDNS)
	} else if *y.HostResolver.Enabled {
		args.UDPDNSLocalPort = udpDNSLocalPort
		args.TCPDNSLocalPort = tcpDNSLocalPort
		args.DNSAddresses = append(args.DNSAddresses, args.SlirpDNS)
	} else if len(y.DNS) > 0 {
		for _, addr := range y.DNS {
			args.DNSAddresses = append(args.DNSAddresses, addr.String())
		}
	} else {
		args.DNSAddresses, err = osutil.DNSAddresses()
		if err != nil {
			return err
		}
	}
```

**Comment by balajiv113** [comment]
@neowu Yes that's the issue. Thanks for the analysis.
Willing to take it for fix ?? Or i can fix it

**Comment by neowu** [comment]
@balajiv113 I am willing to submit PR, though I haven't worked on Go project before :)

just need to make sure the expected behavior, do we always want to use DNS settings if existed, something like
```
	if len(y.DNS) > 0 {
		for _, addr := range y.DNS {
			args.DNSAddresses = append(args.DNSAddresses, addr.String())
		}
	} else if firstUsernetIndex != -1 || *y.VMType == limayaml.VZ {
		args.DNSAddresses = append(args.DNSAddresses, args.SlirpDNS)
	} else if *y.HostResolver.Enabled {
		args.UDPDNSLocalPort = udpDNSLocalPort
		args.TCPDNSLocalPort = tcpDNSLocalPort
		args.DNSAddresses = append(args.DNSAddresses, args.SlirpDNS)
	} else {
		args.DNSAddresses, err = osutil.DNSAddresses()
		if err != nil {
			return err
		}
	}
```

**Comment by balajiv113** [comment]
Fixed as part of #1962
